File: apps/web/src/api/schoolLife.ts
import client from './client';

export interface Incident {
  id: string;
  tenant_id: string;
  student_id: string;
  reported_by: string;
  incident_type: string;
  severity: string;
  description: string;
  action_taken: string | null;
  status: string;
  created_at: string;
}

export interface IncidentCreate {
  student_id: string;
  incident_type?: string;
  severity?: string;
  description: string;
  action_taken?: string;
}

export function getIncidents(params: { student_id?: string; status?: string } = {}) {
  return client.get<Incident[]>('/v1/school-life/incidents', { params });
}

export function createIncident(data: IncidentCreate) {
  return client.post<Incident>('/v1/school-life/incidents', data);
}

export function updateIncident(id: string, data: { action_taken?: string; status?: string; severity?: string }) {
  return client.patch<Incident>(`/v1/school-life/incidents/${id}`, data);
}

File: apps/api/app/db/models/school_life.py
from sqlalchemy import Column, String, Text
from sqlalchemy.dialects.postgresql import UUID

from app.db.base import Base, TenantMixin


class Incident(Base, TenantMixin):
    __tablename__ = "incidents"

    student_id = Column(UUID(as_uuid=True), nullable=False, index=True)
    reported_by = Column(UUID(as_uuid=True), nullable=False)
    incident_type = Column(String(50), nullable=False, default="behavior")  # behavior|absence|late|health|other
    severity = Column(String(20), nullable=False, default="low")  # low|medium|high
    description = Column(Text, nullable=False)
    action_taken = Column(Text, nullable=True)
    status = Column(String(20), nullable=False, default="open")  # open|resolved|escalated
